Added helper methods in validator for the functions in Validation class

lorenzo opened the pull request:
This makes the validation API more discoverable and faster to create
validators. Fixes #7684

lorenzo committed:
Added helper methods in validator for the functions in Validation class

This makes the validation API more discoverable and faster to create
validators. Fixes #7684

lorenzo commented:
I will need a hand for adding the doc blocks to the new methods, who volunteers? :dog:

markstory commented on src/Validation/Validator.php:
```diff
@@ -535,6 +536,324 @@ public function notEmpty($field, $message = null, $when = false)
+    {
+        $extra = array_filter(['on' => $when, 'message' => $message]);
+        return $this->add($field, 'creditCard', $extra + [
+            'rule' => ['cc', $type, true],
```
The deep parameter (3rd param) is not even used in `Validation::luhn()`. I will clean that up separately.

markstory commented on src/Validation/Validator.php:
```diff
@@ -535,6 +536,324 @@ public function notEmpty($field, $message = null, $when = false)
+    {
+        $extra = array_filter(['on' => $when, 'message' => $message]);
+        return $this->add($field, 'utf8Strict', $extra + [
+            'rule' => ['utf8', ['extended' => true]]
```
Extended isn't the same as 'strict'. Perhaps this should be `utf8Extended` or `utf8Emoji`?

lorenzo replied:
Ah! I struggled finding a good name for this, I like your suggestion better

markstory commented:
I can take care of adding API docs.

markstory committed:
Add doc blocks for Validator methods.

Rename utf8Strict to utf8Extended as that better represents what the
method does.

markstory committed:
Add pass/fail test for utf8extended.

markstory committed:
Increase test coverage.

Cover off previously uncovered code paths.

lorenzo committed:
Adding missing return annotations

lorenzo committed:
Fixing some CS errors

lorenzo commented:
Thanks for the help @markstory
